tests/unit/test_diagnostics.py

```diff
@@ -0,0 +1,166 @@
+
+    def test_gaussian(self, seed=0):
+        """Identifying the optimal summary-statistics combination following the Gaussian-noise model.
+
+        Testing the Two-Stage Procedure's implementation for 2-D parameters.
+
+        Parameters
+        ----------
+        seed : int, optional
+
+        """
+        # Defining summary statistics.
+        def ss_uninformative(x):
+            return 1
+        ss_mean = Gauss.ss_mean
+        ss_var = Gauss.ss_var
+
+        list_ss = [ss_mean, ss_var, ss_uninformative]
+
+        # Initialising the simulator.
+        mean_true = 5
+        std_true = 1
+        priors = []
+        priors.append(elfi.Prior('uniform', 0, 10, name='mu'))
+        priors.append(elfi.Prior('truncnorm', 0, 4, name='sigma'))
+
+        fn_simulator = Gauss.gauss
+        y_obs = fn_simulator(mean_true, std_true, random_state=np.random.RandomState(seed))
+        simulator = elfi.Simulator(fn_simulator, *priors, observed=y_obs)
+
+        # Identifying the optimal summary statistics based on the Two-Stage procedure.
+        diagnostics = TwoStageSelection(simulator, 'euclidean', list_ss=list_ss, seed=seed)
+        set_ss_2stage = diagnostics.run(n_sim=1000, n_acc=100, n_closest=20)
+
+        assert ss_uninformative not in set_ss_2stage
+
+    def test_bignk(self, seed=0):
+        """Identifying the optimal summary statistics combination following the bivariate-g-and-k model.
+
+        Testing the Two-Stage Procedure's implementation for 2-D data.
+
+        Parameters
+        ----------
+        seed : int, optional
+
+        """
+        # Defining summary statistics.
+        ss_order = GNK.ss_order
+        ss_robust = GNK.ss_robust
+        ss_octile = GNK.ss_octile
+
+        list_ss = [ss_order, ss_robust, ss_octile]
+
+        # Initialising the simulator.
+        true_params = [3, 4, 1, 0.5, 1, 2, .5, .4, 0.6]
+        priors = []
+        priors.append(elfi.Prior('uniform', 0, 5, name='A1'))
+        priors.append(elfi.Prior('uniform', 0, 5, name='A2'))
+        priors.append(elfi.Prior('uniform', 0, 5, name='B1'))
+        priors.append(elfi.Prior('uniform', 0, 5, name='B2'))
+        priors.append(elfi.Prior('uniform', -5, 10, name='g1'))
+        priors.append(elfi.Prior('uniform', -5, 10, name='g2'))
+        priors.append(elfi.Prior('uniform', -.5, 5.5, name='k1'))
+        priors.append(elfi.Prior('uniform', -.5, 5.5, name='k2'))
+        EPS = np.finfo(float).eps
+        priors.append(elfi.Prior('uniform', -1 + EPS, 2 - 2 * EPS, name='rho'))
+
+        fn_simulator = BiGNK.BiGNK
+        y_obs = fn_simulator(*true_params, random_state=np.random.RandomState(seed))
+        simulator = elfi.Simulator(fn_simulator, *priors, observed=y_obs)
+
+        # Identifying the optimal summary statistics based on the Two-Stage procedure.
+        diagnostics = TwoStageSelection(simulator, euclidean_multiss, list_ss=list_ss, seed=seed)
+        diagnostics.run(n_sim=1000, n_acc=100, n_closest=20)
```

Comment:
`assert` something?

Reply:
It is difficult to say which one will be the best summary statistic so that the test always passes. This is discussed in the g-and-k distribution paper called Likelihood-free Bayesian estimation of multivariate quantile distributions. 

Since we do not assert anything, could mark this test to be ignored by Travis. Though this test case could be useful if the user wanted to familiarise with the summary statistics for the g-and-k distribution-based models.